src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2163,9 +2163,16 @@ private QueryDoneRunning MarkQueryAsRunning(IIndexQuery query, OperationCancelTo
 
         private static bool WillResultBeAcceptable(bool isStale, IndexQueryBase<BlittableJsonReaderObject> query, AsyncWaitForIndexing wait)
         {
+           
             if (isStale == false)
                 return true;
 
+            if (query.WaitForNonStaleResults && query.WaitForNonStaleResultsTimeout == null)
+            {
+                query.WaitForNonStaleResultsTimeout = TimeSpan.FromSeconds(30);
```

Comment:
This should be configurable?
Also this removes the ability to wait indefinitely for the results (by default in C# we are using DefaultTimeout (15sec)).

Reply:
I think to wait 30 seconds It's more then enough and if someone want to wait indefinitely he always can initialize the timeout.
Maybe we can decrease it to DefaultTimeout (15sec)).

Reply:
Should be reduced, yes. And the user can control this by explicitly specifying the timeout.